### ISSUE TITLE
Unable to use `[env:key=value]` directive with `dotnet watch` in 8.0.20x

### ISSUE THREAD

**Issue by xt0rted** [dotnet/sdk#39020]
### Describe the bug

The environment variable directive was removed in 8.0.100 and then in #37827 it was restored. With 8.0.200 and 8.0.201 it still doesn't seem to work, at least not with `dotnet watch` which is where I tend to use this the most.

Neither of these work:

```console
dotnet watch [env:DOTNET_WATCH_RESTART_ON_RUDE_EDIT=true] ...
```

```console
dotnet [env:DOTNET_WATCH_RESTART_ON_RUDE_EDIT=true] watch ...
```

While this does:

```console
dotnet [env:NETCORE_ENVIRONMENT=Development] run ...
```

### To Reproduce

See #37686 for the original report and repro steps.

### Further technical details

<details><summary>dotnet --info</summary>
<p>

```
❯ dotnet --info
.NET SDK:
 Version:           8.0.201
 Commit:            4c2d78f037
 Workload version:  8.0.200-manifests.aa5c7b50

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22621
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\8.0.201\

.NET workloads installed:
There are no installed workloads to display.

Host:
  Version:      8.0.2
  Architecture: x64
  Commit:       1381d5ebd2

.NET SDKs installed:
  6.0.419 [C:\Program Files\dotnet\sdk]
  7.0.116 [C:\Program Files\dotnet\sdk]
  7.0.203 [C:\Program Files\dotnet\sdk]
  7.0.313 [C:\Program Files\dotnet\sdk]
  7.0.406 [C:\Program Files\dotnet\sdk]
  8.0.101 [C:\Program Files\dotnet\sdk]
  8.0.102 [C:\Program Files\dotnet\sdk]
  8.0.201 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.26 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.27 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.26 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.27 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.26 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.27 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.16 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set
```

</p>
</details>

**Comment by baronfel** [comment]
Thanks for the spot - this seems to be because `watch` is an entirely separate binary from the rest of the SDK, with a completely parallel parser/etc. The [root command ](https://github.com/dotnet/sdk/blob/7db679abd7c7292684ead2a7f5ec213dbb8f51c0/src/BuiltInTools/dotnet-watch/CommandLineOptions.cs#L153) for watch needs at least the EnvVarDirective added to it like the ['core' CLI has](https://github.com/dotnet/sdk/blob/4712b35b94f2ad672e69ec35097cf86fc16c2e5e/src/Cli/dotnet/Parser.cs#L20-L23).

cc @tmat - we need to make a plan to either unify these configurations or absorb `watch` into the `dotnet` CLI to prevent gaps like this.

**Comment by tmat** [comment]
@baronfel I'm not sure if it is possible with the current design. Changing the design would break existing scenarios. Unfortunately, it seems like we will need to replicate each new command line feature.

**Comment by tmat** [comment]
Perhaps we should institute a process whereby any new CLI feature needs to be tested with `dotnet watch` before it ships to avoid the gaps.

**Comment by xt0rted** [comment]
Looks like this was fixed in 8.0.204, thanks! 🎉
